dokka-runners/dokka-gradle-plugin/src/main/kotlin/adapters/KotlinAdapter.kt

```diff
@@ -320,8 +376,15 @@ private class KotlinCompilationDetailsBuilder(
                     .artifactView {
                         // Android publishes many variants, which can cause Gradle to get confused,
                         // so specify that we need a JAR and resolve leniently
-                        if (compilation.target.platformType == KotlinPlatformType.androidJvm) {
-                            attributes { artifactType("jar") }
+                        if (compilation.target.platformType == androidJvm) {
+                            attributes { artifactType(artifactType) }
+
+                            // Setting lenient=true is not ideal, because it might hide problems.
```

Comment:
just curious: is it possible to somehow `log` those problems in case of `lenient=true` or are they just silently ignored? just for troubleshooting?

Reply:
It might be possible to log them (`configuration.incoming.artifactView { ... }.artifacts.failures`), but I'd be cautious about using them. It's very easy to accidentally cause Configuration Cache errors with ArtifactView ([example](https://github.com/gradle/gradle/issues/27594)), and I would expect a lot of failures (because this code takes a 'shotgun' approach and just tries to grab a lot of dependencies) so any actual errors could be difficult to differentiate from noise.

I agree logging the failures could be useful though, but I think it'd be best to be careful and check properly.